survey_question_type_five_star/static/src/js/survey.js

```diff
@@ -1,13 +1,44 @@
 /* Copyright 2018 ACSONE SA/NV
 License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).*/
-
-$(document).ready(function () {
+odoo.define('survey_question_type_five_star.survey', function(require) {
     'use strict';
-    $(".rate > label").click(function () {
-        var label_items = $( this ).parent().find("label");
-        var value = label_items.length - $( this ).index();
-        label_items.removeClass("checked");
-        label_items.slice($( this ).index()).addClass("checked");
-        $( this ).parent().find("input").val(value);
+    $(document).ready(function() {
+        //Print Result case
+        var the_form = $('.js_surveyform');
+        var prefill_controller = the_form.attr("data-prefill");
+        if (!the_form.length) {
+            return $.Deferred().reject("DOM doesn't contain '.js_surveyform'");
+        }
+        function prefill() {
+            if (!_.isUndefined(prefill_controller)) {
+                var prefill_def = $.ajax(prefill_controller, {
+                        dataType: "json"
+                    })
+                    .done(function(json_data) {
+                        _.each(json_data, function(value, key) {
+                            var input = the_form.find(".form-control[name=" + key + "]");
+                            $(input).parent().find("label").each(function(label_index) {
+                                if ((5 - label_index) <= value) {
+                                    $(this).addClass("checked");
```

Comment:
We have two different logics to highlight stars, so I think a widget would be better here.